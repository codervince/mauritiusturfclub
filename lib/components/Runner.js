import React from 'react';
import PropTypes from 'prop-types';



const styles = {

  winner: {
    paddingBottom: 5,
    borderBottomStyle: 'solid',
    borderBottomColor:  '#aaa',
    borderBottomWidth:1,
    marginBottom: 10,
    width: '250px',
    border: '1px',
    backgroundColor: '#0f0',
    color: '#000'
  },

  second: {
    backgroundColor: '#ff0',
    color: '#000',
    paddingBottom: 5,
    borderBottomStyle: 'solid',
    borderBottomColor:  '#aaa',
    borderBottomWidth:1,
    marginBottom: 10,
    width: '250px',
    border: '1px'
  },

  third: {
    backgroundColor: '#ffa500',
    color: '#000',
    paddingBottom: 5,
    borderBottomStyle: 'solid',
    borderBottomColor:  '#aaa',
    borderBottomWidth:1,
    marginBottom: 10,
    width: '250px',
    border: '1px'
  },

  race: {
    backgroundColor: '#ccc000',
    color: '#000',
    paddingBottom: 5,
    borderBottomStyle: 'solid',
    borderBottomColor:  '#aaa',
    borderBottomWidth:1,
    marginBottom: 10,
    width: '250px',
    border: '1px',
  },
  raceTitle: {
    fontWeight: 'bold',
  },
  raceDate: {
    fontSize: '0.8em',
    color: '#888'
  },
  runner : {
    paddingTop: 10,
    paddingBottom: 10,

  },
  body: {
    paddingLeft: 20,

  },
  finished: {
    fontSize: '150%',
    color: 'red'
  }
};

const getStyle = (finalPos) => {
  switch(finalPos) {
    case '1':
    case 1:
      return styles.winner;
    case '2':
    case 2:
      return styles.second;
    case '3':
    case 3:
      return styles.third;
    default:
      return styles.race;
  }
};

const dateDisplay = (dateString) => new Date(dateString).toDateString();

const Runner = (props, context) => {
  const {runner} = props;
  let runnerType = getStyle(runner.finalPos);
  // const horse = context.store.lookupHorse(runner.horseName);
  return (
    <div style={runnerType}>
      <div style={styles.raceTitle}> {runner.meetingIndex} {dateDisplay(runner.raceDate)}</div>
      <div style={styles.raceNoRunners}>{runner.noRunners} ran </div>
      <div>No: {runner.horseNo}</div>
      <div>{runner.horseName}</div>
      <div>{runner.jockey}</div>
      <div>{runner.comment}</div>
      <div>Jockey carries: {runner.jockeyWt}</div>
      <div>Horse weighs: {runner.horseWt}</div>
      <div>Barrier: {runner.draw}</div>
      <div>Previous runs: {runner.perf}</div>
      <div style={styles.finished}>Finished: {runner.finalPos}</div>
    </div>
  );
};

// Runner.propTypes = {
//   runner: PropTypes.shape({
//     noRunners: PropTypes.string.isRequired,
//     date: PropTypes.string.isRequired,
//     horseName: PropTypes.string.isRequired,
//     meetingIndex: PropTypes.string.isRequired,
//   })
// };

Runner.contextTypes  = {
  store: PropTypes.object,
};

export default Runner;
