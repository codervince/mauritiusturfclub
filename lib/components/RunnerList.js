import React from 'react';
import Runner from './Runner';

const RunnerList = (props) => {
  return (
    <div>
      {Object.values(props.runners).map( (runner) =>
        <Runner
          key = {runner.uniqueID}
          runner={runner}
          store={props.store}
        />
      )}
    </div>

  );
};
export default RunnerList;
