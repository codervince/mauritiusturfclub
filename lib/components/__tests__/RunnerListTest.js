import React from 'react';
import RunnerList from '../RunnerList';
import 'babel-polyfill';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Runner from '../Runner';

describe('RunnerList', () => {

  const testProps = {
    runners:{
      a: { uniqueID: 'a'},
      b: { uniqueID: 'b'},
    },

  };

  Runner.propTypes = {};

  it('renders correctly', () => {
    const wrapper= shallow(
      <RunnerList
        {...testProps}
      />
    );
    // expect(wrapper.node.props.children.length).toBe(2);
    expect(wrapper.find('Runner').length).toBe(2);
    // expect(tree.children.length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });

});
