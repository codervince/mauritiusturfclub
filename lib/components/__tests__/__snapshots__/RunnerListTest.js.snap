// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunnerList renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Runner
        runner={
            Object {
                "uniqueID": "a",
              }
        }
        store={undefined}
    />
    <Runner
        runner={
            Object {
                "uniqueID": "b",
              }
        }
        store={undefined}
    />
</div>,
  "nodes": Array [
    <div>
      <Runner
            runner={
                  Object {
                        "uniqueID": "a",
                      }
            }
            store={undefined}
      />
      <Runner
            runner={
                  Object {
                        "uniqueID": "b",
                      }
            }
            store={undefined}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <RunnerList
        runners={
                Object {
                        "a": Object {
                          "uniqueID": "a",
                        },
                        "b": Object {
                          "uniqueID": "b",
                        },
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "runners": Object {
            "a": Object {
              "uniqueID": "a",
            },
            "b": Object {
              "uniqueID": "b",
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Runner
                    runner={
                              Object {
                                        "uniqueID": "a",
                                      }
                    }
                    store={undefined}
          />
          <Runner
                    runner={
                              Object {
                                        "uniqueID": "b",
                                      }
                    }
                    store={undefined}
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Runner
                    runner={
                              Object {
                                        "uniqueID": "a",
                                      }
                    }
                    store={undefined}
          />
          <Runner
                    runner={
                              Object {
                                        "uniqueID": "b",
                                      }
                    }
                    store={undefined}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RunnerList
    runners={
        Object {
            "a": Object {
              "uniqueID": "a",
            },
            "b": Object {
              "uniqueID": "b",
            },
          }
    }
/>,
}
`;
